packages/sanity/src/studio/components/navbar/schemaWarnings/SchemaWarningsButton.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useCallback, useMemo, useState} from 'react'
+import {
+  Button,
+  Card,
+  Heading,
+  Text,
+  Menu,
+  MenuButton,
+  Stack,
+  Dialog,
+  DialogProps,
+  Inline,
+} from '@sanity/ui'
+
+import {InfoOutlineIcon, PackageIcon, WarningOutlineIcon} from '@sanity/icons'
+import {useId} from '@reach/auto-id'
+
+import {ChangelogAccordion} from '../changelog/ChangelogAccordion'
+import {isDev} from '../../../../environment'
+import {useSchema} from '../../../../hooks'
+import {useColorScheme} from '../../../colorScheme'
+import {SchemaProblemGroups} from '../../../screens/schemaErrors/SchemaProblemGroups'
+
+export function SchemaWarningsButton() {
+  const schema = useSchema()
+  const {scheme} = useColorScheme()
+  const groupsWithWarnings =
+    schema._validation?.filter((group) =>
+      group.problems.some((problem) => problem.severity === 'warning')
+    ) || []
+
+  const popoverProps = useMemo(
+    () => ({
+      constrainSize: true,
+      portal: true,
+      scheme: scheme,
+    }),
+    [scheme]
+  )
+
+  const dialogId = useId() || 'schema-warnings-menu'
+
+  const [isDialogOpen, setDialogOpen] = useState(false)
+  const [buttonElement, setButtonElement] = useState<HTMLButtonElement | null>(null)
+  const handleOpen = useCallback(() => setDialogOpen(true), [])
+
+  const handleClose = useCallback(() => {
+    setDialogOpen(false)
+
+    if (buttonElement) {
+      buttonElement.focus()
+    }
+  }, [buttonElement])
+
+  const dialogProps: Omit<DialogProps, 'id'> = useMemo(
+    () => ({
+      footer: (
+        <Inline padding={2} space={2}>
+          <InfoOutlineIcon />
+          <Text muted size={1}>
+            Note: this message is only displayed during development
+          </Text>
+        </Inline>
+      ),
+      onClickOutside: handleClose,
+      onClose: handleClose,
+      scheme: 'light',
```

Comment:
Always light? If intentional, can we put a comment here to explain why?

Reply:
Copy paste from the ChangelogDialog - I've asked @hermanwikner if he knows why we need it there - will remove & update with a comment.

Reply:
Ah - I realised now why it's needed: the navbar is dark schemed by default, so the dialog will inherit dark mode from it. That's an interesting issue - I guess what we'd want to express here is more like `scheme: "opposite"` or `scheme: "reset"` or something… 🤔